server/server/Controllers/CartController.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using server.Models;
+using Microsoft.AspNetCore.Authorization;
+using server.Models.ResponseModels;
+using Microsoft.EntityFrameworkCore;
+using server.DBRepositories;
+
+namespace server.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize(Roles = "User")]
+    public class CartController : Controller
+    {
+        private UnitOfWork db;
+        public CartController(ApplicationContext context)
+        {
+            db = new UnitOfWork(context);
+        }
+
+        [HttpGet]
+        [Authorize]
+        public List<ListItem> Get()
+        {
+            string userEmail = HttpContext.User.Identity.Name;
+            List<Cart> userCart = new List<Cart>();
+            userCart = db.Carts.GetAll().Where(c => c.UserId == userEmail).ToList();
+
+            List<ListItem> resultCart = new List<ListItem>();
+            foreach(Cart cartItem in userCart)
+            {
+                Cake cake = db.Catalog.GetById(cartItem.CakeId);//.FirstOrDefault(c => c.Id == cartItem.CakeId);
```

Comment:
Note! there is a query for database inside of loop. It can be bad for performance. Better to retrieve data with one query with filtration on database side.